zilliqa/src/consensus.rs

```diff
@@ -730,7 +784,8 @@ impl Consensus {
 
         // If we haven't applied the transaction yet, do so. This ensures we don't execute the transaction twice if we
         // already executed it in the process of proposing this block.
```

Comment:
OK, so I do not think it makes sense for this function to not execute if it thinks the TX has been executed before. It also represented a bug where when you are proposing a block, it might not be from your HEAD, you can change the state root, but the TX will not execute due to this check.

If the tx has been executed before, it should fail the nonce check.